pkg/kubelet/config/common.go

```diff
@@ -137,10 +135,7 @@ func tryDecodeSinglePod(data []byte, defaultFn defaultFunc) (parsed bool, pod *v
 	if err = defaultFn(newPod); err != nil {
 		return true, pod, err
 	}
-	opts := validation.PodValidationOptions{
-		AllowMultipleHugePageResources: utilfeature.DefaultFeatureGate.Enabled(features.HugePageStorageMediumSize),
-	}
-	if errs := validation.ValidatePodCreate(newPod, opts); len(errs) > 0 {
+	if errs := validation.ValidatePodCreate(newPod, validation.PodValidationOptions{}); len(errs) > 0 {
```

Comment:
Is there anything else passing options to `ValidatePodCreate`? Can we remove this argument entirely? This isn't how I've normally seen feature flags used in the kubelet.

Reply:
Yes, there are other gated features that use PodValidationOptions, e.g. AllowDownwardAPIHugePages, AllowIndivisibleHugePagesValues etc. We can't remove it.

Reply:
Confirmed: https://github.com/kubernetes/kubernetes/blob/31f6cca256a21b9bbd561965dd321f518c88c944/pkg/apis/core/validation/validation_test.go#L4287

Reply:
yeah, this has been a useful pattern for injection validation options for similar changes to pod spec evolution.